src/workerd/jsg/setup.h

```diff
@@ -224,6 +236,25 @@ class IsolateBase {
   // use `->InstanceTemplate()->NewInstance()` to construct an object, and you can pass this to
   // `FindInstanceInPrototypeChain()` on an existing object to check whether it was created using
   // this template.
+
+  static void promiseHook(v8::PromiseHookType type,
+                          v8::Local<v8::Promise> promise,
+                          v8::Local<v8::Value> parent);
+  void pushAsyncFrame(AsyncContextFrame& next);
+  // Pushes the frame onto the stack making it current. Importantly, the stack
+  // does not maintain a refcounted reference to the frame so it is important
+  // for the caller to ensure that the frame is kept alive.
+  void popAsyncFrame();
+
+  std::deque<AsyncContextFrame*> asyncFrameStack;
```

Comment:
Why `std::deque` and not `std::stack` or even `kj::Vector`?

Reply:
No specific reason really. Just mostly used to using `deque` ;-) ... I'll switch to `kj::Vector`